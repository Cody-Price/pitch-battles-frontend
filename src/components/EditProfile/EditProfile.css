.edit-profile-form {
  display: grid;
  grid-template-columns: 130px 1fr;
  animation: account-subsection-fade-in 0.3s linear;
}

.edit-profile-form *:focus {
  outline: none;
}

.profile-submit-btn {
  grid-column: 1/3;
  width: 150px;
  height: 23px;
  font-family: "chronotype";
  font-size: 1rem;
  margin: 1rem auto;
  background: #6a9dd4;
  border: 0;
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.7);
}

.profile-submit-btn:active {
  box-shadow: none;
}

.edit-profile-form input {
  font-family: "chronotype";
  font-size: 1rem;
  height: 18px;
  width: 300px;
  margin-top: 1rem;
  background: #6a9dd4;
  border: 0;
  padding-left: 8px;
}

.edit-profile-form p {
  padding-top: 1.3rem;
  margin: 0;
}

@keyframes account-subsection-fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@media (max-width: 770px) {
  .edit-profile-form {
    padding-left: 1rem;
    display: block;
    grid-template-columns: 130px 1fr;
    animation: account-subsection-fade-in 0.3s linear;
  }

  .edit-profile-form input {
    width: 140px;
    margin-top: 0;
  }

  .profile-submit-btn {
    margin: 1rem 0rem;
    grid-column: 1;
  }
}
