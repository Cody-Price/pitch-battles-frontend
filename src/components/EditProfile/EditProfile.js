import React, { Component } from "react";
import "./EditProfile.css";

class EditProfile extends Component {
  constructor() {
    super();

    this.state = {
      first_name: "",
      last_name: ""
    };
  }

  componentDidMount() {
    this.populateOnLoad();
  }

  populateOnLoad = () => {
    this.setState({
      first_name: this.props.user.first_name,
      last_name: this.props.user.last_name
    });
  };

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = event => {
    event.preventDefault();

    if (this.state.first_name === "" && this.state.last_name === "") {
      return;
    }
    const update = {
      first_name: this.state.first_name,
      last_name: this.state.last_name
    };

    this.props.changeProfile(update);
  };

  render() {
    return (
      <section className="edit-profile">
        <form
          className="edit-profile-form"
          onSubmit={event => {
            this.handleSubmit(event);
          }}
        >
          <p className="first-name-edit-label">first name:</p>
          <input
            name="first_name"
            className="profile-first-name-input"
            value={this.state.first_name}
            placeholder="first name"
            onChange={event => {
              this.handleChange(event);
            }}
          />
          <p className="last-name-edit-label">last name:</p>

          <input
            name="last_name"
            className="profile-last-name-input"
            value={this.state.last_name}
            placeholder="last name"
            onChange={event => {
              this.handleChange(event);
            }}
          />

          <button type="submit" className="profile-submit-btn">
            submit new info
          </button>
        </form>
      </section>
    );
  }
}

export default EditProfile;
