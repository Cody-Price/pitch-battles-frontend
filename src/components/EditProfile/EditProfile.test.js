import React from "react";

import EditProfile from "./EditProfile";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

jest.mock("../../utilities/fetchCalls");

describe("EditProfile", () => {
  let wrapper;
  const mockUser = {
    id: 12,
    attributes: {
      first_name: "Haley",
      last_name: "Jacobs"
    }
  };

  const mockWebToken = "123123124";

  beforeEach(() => {
    wrapper = shallow(<EditProfile user={mockUser} webToken={mockWebToken} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("populateOnLoad", () => {
    it("should setState with props", () => {
      const expected = {
        error: false,
        first_name: "Haley",
        last_name: "Jacobs",
        success: false
      };

      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe("handleChange", () => {
    it("should call handleChange on change from source 1 with the correct params", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleChange");
      const expected = {
        target: {
          name: "first_name",
          value: "Steve"
        }
      };
      wrapper.instance().forceUpdate();

      wrapper.find(".profile-first-name-input").simulate("change", expected);

      expect(spy).toHaveBeenCalledWith(expected);
    });

    it("should call handleChange on change from source 2 with the correct params", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleChange");
      const expected = {
        target: {
          name: "last_name",
          value: "Moore"
        }
      };
      wrapper.instance().forceUpdate();

      wrapper.find(".profile-last-name-input").simulate("change", expected);

      expect(spy).toHaveBeenCalledWith(expected);
    });

    it("should setState", () => {
      const mockEvent = {
        target: {
          name: "first_name",
          value: "Steve"
        }
      };

      const expected = {
        error: false,
        first_name: "Steve",
        last_name: "Jacobs",
        success: false
      };

      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe("handleSubmit", () => {
    let mockPreventDefault;

    it("should return if state is empty strings", () => {
      mockPreventDefault = jest.fn();
      const mockEvent = { preventDefault: mockPreventDefault };
      const spy = jest.spyOn(wrapper.instance(), "handleSubmit");

      wrapper.setState({
        first_name: "",
        last_name: ""
      });

      wrapper.instance().forceUpdate();

      wrapper.instance().handleSubmit(mockEvent);

      expect(spy).toReturn();
    });

    it("should call handleSubmit on submit", () => {
      mockPreventDefault = jest.fn();
      const mockEvent = { preventDefault: mockPreventDefault };

      const spy = jest.spyOn(wrapper.instance(), "handleSubmit");
      wrapper.setState({
        first_name: "Kevin",
        last_name: "Simpson"
      });

      wrapper.find(".edit-profile-form").simulate("submit", mockEvent);

      expect(spy).toHaveBeenCalled();
    });

    it("should call changelProfileFetch", () => {
      const spy = jest.spyOn(wrapper.instance(), "changeProfileFetch");
      mockPreventDefault = jest.fn();
      const mockEvent = { preventDefault: mockPreventDefault };

      wrapper.instance().handleSubmit(mockEvent);

      expect(spy).toHaveBeenCalled();
    });
  });
});
