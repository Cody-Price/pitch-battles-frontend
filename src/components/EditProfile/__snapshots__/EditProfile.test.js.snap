// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfile should match the snapshot 1`] = `
<section
  className="edit-profile"
>
  <form
    className="edit-profile-form"
  >
    <input
      className="profile-first-name-input"
      name="first_name"
      placeholder="first name"
      value="Haley"
    />
    <input
      className="profile-last-name-input"
      name="last_name"
      placeholder="last name"
      value="Jacobs"
    />
    <input
      className="profile-email-input"
      name="email"
      placeholder="email"
      type="email"
      value="hjacobs@gmail.com"
    />
    <button
      className="profile-submit-btn"
      type="submit"
    >
      submit new info
    </button>
  </form>
</section>
`;
